### DISCUSSION TITLE
[1 febbraio 2022 - A] Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#206] [Esami]
**Esercizio 2 (1+1+1+2+2+1 punti)**

La funzione di 4 variabili, 𝑓(a,b,c,d), vale 0:

- se $a(b \oplus c)=1$
- se $a+b+d=0$

---

- Si stenda la tavola di verità di _f_
- Si scrivano le espressioni canoniche SOP e POS di _f_
- Si scrivano le espressioni minimali SOP e POS di _f_
- Si realizzi la funzione _f_ utilizzando un MUX ad 4 ingressi
- Si scriva l’espressione di _f_ in forma ALL-NAND

**Comment by Elia-Belli** [comment]
- $Y = \bar a \bar b \bar c d + \bar a \bar b cd + \bar a b \bar c \bar d + \bar a b \bar c d + \bar a bc \bar d + \bar a bcd + a \bar      b \bar c \bar d + a \bar b \bar c d + abc \bar d + abcd$ (SOP canonica)
  $Y = (a+b+c+d)(a+b+\bar c + d)(\bar a + b + \bar c + d)(\bar a +b + \bar c + \bar d)(\bar a + \bar b + c +d)(\bar a +\bar b + c + \bar d)$ (POS canonica)
- $Y = \bar a b + \bar a d + bc + a \bar b \bar c$ (SOP minima)
  $Y = (a+b+d)(\bar a + \bar b + c)(\bar a + b + \bar c)$ (POS minima)
- $Y =\overline {\overline {\bar a b + \bar a d + bc + a \bar b \bar c}} = \overline {(\overline {\bar a b})(\overline {\bar a d})(\overline {bc})(\overline {a \bar b \bar c})}$ (ALL-NAND)
![k](https://user-images.githubusercontent.com/107944476/212468952-b0a8322d-db65-4993-9734-91a4b3d8626c.png)

**Comment by Colasuonno** [comment]
Ciao! L'unica cosa che non mi torna è la forma di f scritta ALL-NAND, probabilmente è md che non aggiunge i (doppi) negati sopra?

**Comment by Elia-Belli** [comment]
@Colasuonno non saprei, comunque per arrivare a quella forma ho negato 2 volte la forma minima SOP e poi ho svolto uno dei due negati con De Morgan.

**Comment by fraxb04** [comment]
@Elia-Belli 
io la forma all-NAND la scriverei così:
![image](https://user-images.githubusercontent.com/122573871/216781273-3595f03c-1d7f-416d-b00b-161e4cdeb2e7.png)

**Comment by Elia-Belli** [comment]
@fraxb04 Nessuna di quelle è una forma $ALL-NAND$
La prima riga è un $NOR$ di $NAND$
La seconda un $NOR$ di $OR$
La terza un $ALL-AND$
Non so come hai ottenuto quelle forme e nella seconda ti verrebbe addirittura $0$: siccome $OR$ è associativo hai termini come $A + \bar A$ che danno $1$, se neghi tutto $0$.

**Comment by fraxb04** [comment]
@Elia-Belli 
quindi in generale quale è la formula per trovarsi un'all-NAND?

**Comment by Elia-Belli** [comment]
@fraxb04 Neghi 2 volte la $SOP$ e svolgi un negato con De Morgan

**Comment by fraxb04** [comment]
@Elia-Belli 
ok ho capito cosa intendi, di base non esiste una formula ma bisogna cercare di ottenere ovviamente come dice la parola una forma totalmente NAND.
Grz

**Comment by M1SC0** [comment]
![image](https://github.com/user-attachments/assets/fc96a1ac-6e5e-4008-9057-9e25d53a74ea)
